### PR TITLE
Fix #619: commented out some of the console logs

### LINKED ISSUE #619: Remove unnecessary console logging in LiveDevelopment code
The LiveDev code spams the console a lot, and mostly for unnecessary things.  I think it was added originally to make debugging easier.  It isn't necessary now, so let's remove as much of it as makes sense:

Running `git grep "console.log" src/LiveDevelopment | grep -v "node_modules"` I see the following as some examples (NOTE: there are some we don't care about, since we use the MultiBrowser code paths):

```
src/LiveDevelopment/Documents/HTMLDocument.js:                    console.log(delta);
src/LiveDevelopment/Documents/HTMLDocument.js:            console.log("Edits applied to browser were:");
src/LiveDevelopment/Documents/HTMLDocument.js:            console.log(JSON.stringify(result.edits, null, 2));
src/LiveDevelopment/Inspector/Inspector.js:            console.log("You must connect to the WebSocket before sending messages.");
src/LiveDevelopment/LiveDevMultiBrowser.js:            console.log("Invalid launcher object: ", launcher, new Error("LiveDevMultiBrowser.setLauncher()"));
src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js:            console.log("Edits applied to browser were:");
src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js:            console.log(JSON.stringify(result.edits, null, 2));
src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js:                console.log("[Brackets LiveDev] Received message without method.");
src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js:                        console.log("[Brackets LiveDev] Error executing a handler for " + msg.method);
src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js:                        console.log(e.stack);
src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js:                console.log("[Brackets LiveDev] No subscribers for message " + msg.method);
src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js:                console.log("[Brackets LiveDev] Trying to send a response for a message with no ID");
src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js:            console.log("Runtime.evaluate");
src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js:                console.log("[Brackets LiveDev] Malformed message received: ", msgStr);
```

The ones that are error handling cases are probably OK.  It's the ones that run by default.

### PR DESCRIPTION
**Pull request by Abykin**
fixes #619
Some console.log in livedevolopment were being ran a lot of time. Here is a summary of which logs were commented out. I felt like the ones that are used for error handling can be left untouched as they were not causing unnecessary console spam. Let me know if I missed something or any additional changes should be made.

**HTMLDocument.js**
Removed		console.log(delta);
Removed          console.log("Edits applied to browser were:");
Removed            console.log(JSON.stringify(result.edits, null, 2));
**Inspector.js**
Removed   console.log("You must connect to the WebSocket before sending messages.");
**LiveDevMultiBrowser.js**
Kept console.log("Invalid launcher object: ", launcher, new Error("LiveDevMultiBrowser.setLauncher()"));
**LiveHTMLDocument.js**
Removed console.log("Edits applied to browser were:");
Removed console.log(JSON.stringify(result.edits, null, 2));
**LiveDevProtocolRemote.js**
Kept console.log("[Brackets LiveDev] Received message without method.");
Kept console.log("[Brackets LiveDev] Error executing a handler for " + msg.method);
Kept console.log(e.stack);
Removed console.log("[Brackets LiveDev] No subscribers for message " + msg.method);
Kept console.log("[Brackets LiveDev] Trying to send a response for a message with no ID");
Removed console.log("Runtime.evaluate");
Kept console.log("[Brackets LiveDev] Malformed message received: ", msgStr);

### CODE REVIEW

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 1]
```diff
@@ -807,6 +807,7 @@ define(function (require, exports, module) {
      */
     function setLauncher(launcher) {
         if (!(launcher && launcher.launch)) {
+            // Useful for debugging on a condition where launcher failed
```
Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: Let's just revert all the changes to this file, since none of them are really needed.

File: src/LiveDevelopment/Documents/HTMLDocument.js [hunk 2]
```diff
@@ -326,8 +327,9 @@ define(function HTMLDocumentModule(require, exports, module) {
         // edit this file or set a conditional breakpoint at the top of this function:
         //     "this._debug = true, false"
         if (this._debug) {
```
Comment (humphd) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: Why don't we just put `this._debug = false;` near the top of this file and leave the rest of this in place.

File: src/LiveDevelopment/Documents/HTMLDocument.js [hunk 3]
```diff
@@ -263,7 +263,8 @@ define(function HTMLDocumentModule(require, exports, module) {
                 console.warn("Browser DOM does not match after change: " + JSON.stringify(change));
 
                 edits.forEach(function (delta) {
```
Comment (humphd) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: Let's just get rid of this entire loop, since we're doing no work, there's no point spending cycles on it.

Comment (humphd) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: On second thought, wrap the whole thing in an `if(this._debug) {...` so it only happens if we are debugging.

File: src/LiveDevelopment/Inspector/Inspector.js [hunk 4]
```diff
@@ -116,7 +116,8 @@ define(function Inspector(require, exports, module) {
      */
     function _send(method, signature, varargs) {
         if (!_socket) {
-            console.log("You must connect to the WebSocket before sending messages.");
```
Comment (humphd) [inline src/LiveDevelopment/Inspector/Inspector.js]: You can revert this change, since we don't ever use this code path in the Browser.

File: src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js [hunk 5]
```diff
@@ -231,8 +231,9 @@ define(function (require, exports, module) {
         // edit this file or set a conditional breakpoint at the top of this function:
         //     "this._debug = true, false"
         if (this._debug) {
```
Comment (humphd) [inline src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js]: Similar to above, let's do `this._debug = false;` somewhere above and have this get skipped in release builds, but we can enable later if we need it.

File: src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js [hunk 6]
```diff
@@ -84,7 +84,7 @@
             } else {
                 // no subscribers, ignore it.
                 // TODO: any other default handling? (eg. specific respond, trigger as a generic event, etc.);
-                console.log("[Brackets LiveDev] No subscribers for message " + msg.method);
+                //console.log("[Brackets LiveDev] No subscribers for message " + msg.method);
```
Comment (humphd) [inline src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js]: This one you can leave, let's revert it.

File: src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js [hunk 7]
```diff
@@ -139,7 +139,8 @@
          * Evaluate an expresion and return its result.
          */
         evaluate: function (msg) {
-            console.log("Runtime.evaluate");
```
Comment (humphd) [inline src/LiveDevelopment/MultiBrowserImpl/protocol/remote/LiveDevProtocolRemote.js]: Just delete it outright, and get rid of your comments.

**Comment by Abykin** [comment]
@humphd Applied your suggested changes in the new commit

File: src/LiveDevelopment/Documents/HTMLDocument.js [hunk 8]
```diff
@@ -63,7 +63,7 @@ define(function HTMLDocumentModule(require, exports, module) {
 
         this.editor = editor;
         this._instrumentationEnabled = false;
-
+        this._debug = false;
```
Comment (humphd) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: Add a comment above this:

```
// XXXBramble: we force debug false to skip various console.log()s below.
```

File: src/LiveDevelopment/Documents/HTMLDocument.js [hunk 9]
```diff
@@ -261,10 +261,12 @@ define(function HTMLDocumentModule(require, exports, module) {
 
             if (edits.length > 0) {
                 console.warn("Browser DOM does not match after change: " + JSON.stringify(change));
-
-                edits.forEach(function (delta) {
-                    console.log(delta);
-                });
+                if(this._debug){
```
Comment (humphd) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: Move this check to 261 above.

File: src/LiveDevelopment/Documents/HTMLDocument.js [hunk 10]
```diff
@@ -261,10 +261,12 @@ define(function HTMLDocumentModule(require, exports, module) {
...
+                  edits.forEach(function (delta) {
+                    // Uncomment to see all delta values (issue-619)
+                    //console.log(delta);
```
Comment (humphd) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: You can uncomment this and remove the comment above, since it's surrounded by the `if(_debug)` check now.

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 11]
```diff
@@ -91,8 +91,8 @@ define(function (require, exports, module) {
 
     // XXXBramble: we need to get this loaded so our extension can require/use it later
     require("LiveDevelopment/Servers/BaseServer");
-
```
Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: Revert all changes to this file (`git checkout master src/LiveDevelopment/LiveDevMultiBrowser.js`)

File: src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js [hunk 12]
```diff
@@ -57,7 +57,7 @@ define(function (require, exports, module) {
             stylesheets: {},
             scripts: {}
         };
-
+        this._debug = false;
```
Comment (humphd) [inline src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js]: Same comment above this line.

File: src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js [hunk 13]
```diff
@@ -204,7 +204,7 @@ define(function (require, exports, module) {
             result              = HTMLInstrumentation.getUnappliedEditList(this.editor, change),
             applyEditsPromise,
             reloadPromise;
-        
+
```
Comment (humphd) [inline src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js]: Revert this please.

File: src/LiveDevelopment/Documents/HTMLDocument.js (hunk 8 above)
Comment (Pomax) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: I'm not familiar with the `XXXBramble` prefix, is this a brackets convention or us?

Comment (humphd) [inline src/LiveDevelopment/Documents/HTMLDocument.js]: Discussed on Mattermost, but I'll repeat here for searchability.  When we change things in the core Brackets code in ways that are only needed for Bramble, we've tended to add a `XXXBramble` comment, so as to make it clear why we're doing this at all.  It's not critical, but it's helpful when we do big upstream merges later.

**Comment by Pomax** [comment]
@Abykin can you take out "Fix for issue-619:" from the title for this PR, and instead add a line to the PR comment that says "fixes #619"? That way github can set up an automatic crosslink between the issue and the associated PR

(when you see "fixes ..." that's not just a helpful hint by whoever filed the PR of which the exact phrasing is up to whoever does the writing,  it's actually very specific github.com syntax for setting up crosslinks =)

**Comment by Abykin** [comment]
@Pomax Changed it, hopefully that worked.

**Comment by humphd** [comment]
@Abykin let me know when you've fixed this and I can review again.

**Comment by Abykin** [comment]
@humphd I have just one question about reverting the line change in LiveHTMLDocument.js. Is there a git command I can run, or should I just manually replace it?

**Comment by humphd** [comment]
If you're reverting changes to an entire file, use `git checkout master <filename>`, and it will get just the file you specify from `master` vs. checking out all of master.

**Comment by Abykin** [comment]
What about reverting just a single change in the file, is that possible with a command?

**Comment by humphd** [comment]
You can try doing https://makandracards.com/makandra/843-git-interactive-reverts, but you might find it easier to just do it manually.  If this turns into a huge problem, don't worry about it.  It's just a good practice to get into, but it's not a deal breaker.

**Comment by Abykin** [comment]
@humphd Pushed the requested changes in the new commit

**Comment by humphd** [comment]
Works great, merging!  Thanks a lot @Abykin.
